src/aap_eda/core/utils/k8s_service_name.py

```diff
@@ -0,0 +1,47 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import re
+
+
+class InvalidRFC1035Label(Exception):
+    pass
+
+
+def create_k8s_service_name(activation_name: str) -> str:
+    """Attempt to convert the activation name to a 1035 name.
+
+    Convert dot(.) underscore(_) space( ) to - and convert it to lowercase
+    """
+    name = re.sub(r"[._\s+]", "-", activation_name.lower())
+    if not rfc_1035_complicant(name):
+        raise InvalidRFC1035Label(
+            "Please provide a service name, that's a valid RFC 1035 name, "
+            "couldn't auto generate service name from activation name"
+        )
+
+    return name
+
+
+def rfc_1035_complicant(name: str) -> bool:
+    """Check if name is compatible with RFC 1035.
+
+    - contain at most 63 characters
+    - contain only lowercase alphanumeric characters or '-'
+    - start with an alphabetic character
+    - end with an alphanumeric character
+    """
+    qualified_pattern = re.compile(r"^[a-z]([a-z-0-9-]{0,61}[a-z])?$")
```

Comment:
1. The hyphen - is specified twice, is redundant but not harmful.
2. RFC 1035 allows labels to end with digits. 

I suggest: 
`qualified_pattern = re.compile(r"^[a-z]([a-z0-9-]{0,61}[a-z0-9])?$")`